syntax = "proto3";
package collector_service;

option go_package = "go.easyops.local/contracts/protorepo-models/easyops/model/collector_service";





/*
聚合指标规则
*/
message CollectorAggregateMetricRule {
message AggregateVars {
    /*
    变量名
    */
    string name = 1 ;
    /*
    指标名
    */
    string metric = 2 ;
}
        
    
    /*
    实例Id
    */
    string instanceId = 1 ;
    /*
    聚合规则名称
    */
    string name = 2 ;
    /*
    聚合指标名
    */
    string metricName = 3 ;
    /*
    聚合指标描述
    */
    string metricDesc = 4 ;
    /*
    聚合指标所属模型id
    */
    string metricOwner = 5 ;
    /*
    聚合指标数据来源模型id
    */
    string metricSource = 6 ;
    /*
    关系路径id
    */
    string pathId = 7 ;
    /*
    关系路径查询
    */
    string path = 8 ;
    /*
    指标数据类型
    */
    string aggregateFunc = 9 ;
    /*
    聚合表达式
    */
    string aggregateExpr = 10 ;
    /*
    聚合变量
    */
    repeated AggregateVars aggregateVars = 11 ;
    /*
    聚合粒度
    */
    int32 granularity = 12 ;
    /*
    分组字段
    */
    repeated string groupBy = 13 ;
}

